src/D2L.CodeStyle.Analyzers/CommonFixes/AddAttributeCodeFix.cs

```diff
@@ -46,9 +48,16 @@ CodeFixContext context
 				.ConfigureAwait( false ) as CompilationUnitSyntax;
 
 			foreach( var diagnostic in context.Diagnostics ) {
-				var identifierSpan = diagnostic.Location.SourceSpan;
+				// Sometimes the diagnostic is on the declaration that needs the attribute,
+				// if not it is the last "additional location" by convention.
+				var location = diagnostic.AdditionalLocations?.LastOrDefault() ?? diagnostic.Location;
 
-				var decl = root.FindNode( identifierSpan ) as TypeDeclarationSyntax;
+				var node = root.FindNode( location.SourceSpan );
+				var decl = node as MemberDeclarationSyntax;
+
+				if( decl == null ) {
+					throw new NotImplementedException( $"Adding attributes to {node.Kind()} is not supported. This is a D2L.CodeStyle.Analyzers bug." );

```

Comment:
Should only happen because we add a new feature but don't do anything here first.